components/predict-form/PredictProvider.tsx

```diff
@@ -0,0 +1,181 @@
+          path: ["options"],
+        })
+      }
+    }
+  })
+
+export type PredictFormType = z.infer<typeof unifiedPredictFormSchema>
+
+function usePredictFormStandalone() {
+  const [questionType, setQuestionType] = useState<QuestionType>(
+    QuestionType.BINARY,
+  )
+  // `setQuestionInFocus` is manually called via `onFocus`, `onBlur` callbacks,
+  // so this could be prone to getting out of sync
+  const [questionInFocus, setQuestionInFocus] = useState(false)
+
+  const form = useForm<PredictFormType>({
+    mode: "all",
+    resolver: zodResolver(unifiedPredictFormSchema),
+  })
+
+  const questionTypeChangedCallback = useCallback(
+    (newType: QuestionType) => {
+      const currentValues = form.getValues()
+      if (newType === QuestionType.MULTIPLE_CHOICE) {
+        form.reset(
+          {
+            ...currentValues,
+            options: currentValues.options?.length
+              ? currentValues.options
+              : [
+                  { text: "", forecast: NaN },
+                  { text: "", forecast: NaN },
+                ],
+            predictionPercentage: undefined,
+          },
+          { keepDefaultValues: true },
+        )
+      } else {
+        form.reset(
+          {
+            ...currentValues,
+            predictionPercentage: currentValues.predictionPercentage ?? NaN,
+            options: undefined,
+          },
+          { keepDefaultValues: true },
+        )
+      }
+    },
+    [form],
+  )
+
+  useEffect(() => {
+    questionTypeChangedCallback(questionType)
+  }, [questionType, questionTypeChangedCallback])
+
+  return {
+    ...form,
+    questionType,
+    setQuestionType,
+    questionInFocus,
+    setQuestionInFocus,
+  }
+}
+
+// eslint-disable-next-line @typescript-eslint/naming-convention
+const PredictFormContext = createContext<ReturnType<
+  typeof usePredictFormStandalone
+> | null>(null)
+
+export function usePredictForm() {
+  const contextForm = useContext(PredictFormContext)
+  // This is inefficient in the case where `contextForm` exists because it still instantiates `standaloneForm`,
+  // additionally it could result in some surprising behaviour if `contextForm` initially exists and is then set to null.
+  // It's hard to avoid this double instantiation because conditionally calling hooks is not allowed.
+  const standaloneForm = usePredictFormStandalone()
+
+  return contextForm ?? standaloneForm
+}
+
+/**
+ * Optional provider that exposes the contents of a nested `<Predict />` form to all child components.
+ *
+ * Example usage:
+ * ```
+ * <PredictProvider>
+ *   <Predict />
+ *   <SiblingComponent />
+ * </PredictProvider>
+ * ```
+ * ...
+ * ```
+ * function SiblingComponent() {
+ *   // Retrieves the same instance as <Predict />, so <SiblingComponent />
+ *   // can interact with the form
+ *   const form = usePredictForm()
+ * }
+ * ```
+ */
+export function PredictProvider({ children }: { children: ReactNode }) {
```

Comment:
Worth reading this comment, this is how I'm allowing the onboarding checklist and `<Predict>` form to control each other